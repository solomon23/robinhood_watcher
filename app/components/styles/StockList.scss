.stockList {
  overflow-y: scroll;
  background-color: #303032;
  padding: 4px 16px;
  margin: 8px;
  flex: 1 1 auto;
}
