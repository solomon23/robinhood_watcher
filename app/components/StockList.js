// @flow
import React, { Component } from 'react'
import StockRow from './StockRow'
import styles from './styles/StockList.scss'

type Props = {
  stocks: Array
}

export default class StockList extends Component<Props> {
  props: Props

  render() {
    const { stocks } = this.props

    return (
      <div className={styles.stockList}>
        {stocks.sort((a, b) => a.symbol.localeCompare(b.symbol)).map(stock => (
          <StockRow key={stock.symbol} stock={stock} />
        ))}
      </div>
    )
  }
}
