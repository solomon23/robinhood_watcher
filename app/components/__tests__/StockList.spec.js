import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import StockList from '../StockList'

Enzyme.configure({ adapter: new Adapter() })

let actions = null

const defaultState = {
  stocks: [],
}

describe('StockList component', () => {
  beforeEach(() => {
    actions = { }
  })

  it('should display empty', () => {
    const component = shallow(<StockList {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should sort list', () => {
    const state = { ...defaultState, stocks: [{ symbol: 'z' }, { symbol: 'a' }, { symbol: 'c' }] }
    const component = shallow(<StockList {...state} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should show as watchlist', () => {
    const state = { ...defaultState, stocks: [{ symbol: 'z' }, { symbol: 'a' }, { symbol: 'c' }] }
    const component = shallow(<StockList {...state} {...actions} watchlist />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
