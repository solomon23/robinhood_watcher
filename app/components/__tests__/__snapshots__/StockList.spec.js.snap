// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockList component should display empty 1`] = `
<div
  className="stockList"
/>
`;

exports[`StockList component should sort list 1`] = `
<div
  className="stockList"
>
  <StockList
    key="a"
    stock={
      Object {
        "symbol": "a",
      }
    }
  />
  <StockList
    key="c"
    stock={
      Object {
        "symbol": "c",
      }
    }
  />
  <StockList
    key="z"
    stock={
      Object {
        "symbol": "z",
      }
    }
  />
</div>
`;
